lib/agent0/interactive_fuzz/fuzz_path_independence.py

```diff
@@ -104,26 +114,39 @@
             agent.close_short(maturity_time=trade.maturity_time, bonds=trade.bond_amount)
 
     # Check the reserve amounts; they should be unchanged now that all of the trades are closed
-    pool_state = interactive_hyperdrive.hyperdrive_interface.get_hyperdrive_state()
-    assert (
-        start_share_reserves == pool_state.pool_info.share_reserves
-    ), f"{start_share_reserves=} != {pool_state.pool_info.share_reserves}"
-    assert (
-        start_shorts_outstanding == pool_state.pool_info.shorts_outstanding
-    ), f"{start_shorts_outstanding=} != {pool_state.pool_info.shorts_outstanding}"
-    assert (
-        start_withdraw_pool_proceeds == pool_state.pool_info.withdrawal_shares_proceeds
-    ), f"{start_withdraw_pool_proceeds=} != {pool_state.pool_info.withdrawal_shares_proceeds}"
-    assert (
-        start_share_price == pool_state.pool_info.share_price
-    ), f"{start_share_price=} != {pool_state.pool_info.share_price}"
-    assert (
-        start_global_exposure == pool_state.pool_info.long_exposure
-    ), f"{start_global_exposure=} != {pool_state.pool_info.long_exposure}"
-    assert (
-        start_hyperdrive_balance == pool_state.hyperdrive_base_balance
-    ), f"{start_hyperdrive_balance=} != {pool_state.hyperdrive_base_balance}"
-    assert (
-        start_gov_fees_accrued == pool_state.gov_fees_accrued
-    ), f"{start_gov_fees_accrued=} != {pool_state.gov_fees_accrued}"
-    assert start_total_shares == pool_state.vault_shares, f"{start_total_shares=} != {pool_state.vault_shares}"
+
+    pool_info_df = interactive_hyperdrive.get_pool_info(coerce_float=False)
+
+    # On first run, save final state
+    if first_run:
+        check_data["check_pool_info_df"] = pool_info_df[check_columns].iloc[-1].copy()
+        # TODO add these to pool info
```

Comment:
we don't add anything to pool info on purpose, the `pool_info` fields should exactly reflect what the contract returns. The `pool_state` container is used to hold any additional variables, like `vault_shares`

Reply:
Fixed in https://github.com/delvtech/agent0/pull/1153